ggu/main.c

```diff
@@ -16,11 +16,390 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdbool.h>
+
+#include <rte_ip.h>
+#include <rte_udp.h>
+#include <rte_log.h>
+#include <rte_lcore.h>
+#include <rte_malloc.h>
+#include <rte_ethdev.h>
+#include <rte_atomic.h>
+
+#include "gatekeeper_gk.h"
 #include "gatekeeper_ggu.h"
+#include "gatekeeper_net.h"
+#include "gatekeeper_main.h"
+#include "gatekeeper_config.h"
+
+#define GGU_PD_VER1 (1)
+
+static void
+process_single_policy(const struct ggu_policy *policy, const struct ggu_config *ggu_conf)
+{
+	struct gk_cmd_entry *entry;
+	/*
+	 * Obtain mailbox of that GK block,
+	 * and send the policy decision to the GK block.
+	 */
+	struct mailbox *mb =
+		get_responsible_gk_mailbox(&policy->flow, ggu_conf->gk);
+
+	if (mb == NULL)
+		return;
+
+	entry = (struct gk_cmd_entry *)mb_get_entry(mb);
```

Comment:
The cast is not needed. It's equivalent to the `malloc()` case, and shouldn't produce any warning.